jax/numpy/lax_numpy.py

```diff
@@ -897,6 +897,46 @@ def histogram(a, bins=10, range=None, weights=None, density=None):
     counts = counts / bin_widths / counts.sum()
   return counts, bin_edges
 
+_range = range
+@_wraps(np.histogramdd)
+def histogramdd(sample, bins=10, range=None, weights=None, density=None):
+  sample = asarray(sample)
+  N, D = shape(sample)
+
+  if weights is not None and weights.shape != (N,):
+    raise ValueError("should have one weight for each sample.")
+
+  bin_idx_by_dim = D*[None]
+  nbins = D*[None]
+  bin_edges_by_dim = D*[None]
+  dedges = D*[None]
+
+  for i in _range(D):
```

Comment:
Would it be possible to call `histogram()` on each dimension, and then combine the results? It seems like this duplicates much of the logic spelled-out there.

Reply:
That's what I was hoping to do initially.  The problem is that we combine across dimensions based on the bin index for each example - unfortunately `histogram()` only returns the counts so the bin indexes would get lost.  [Numpy's implementation ](https://github.com/numpy/numpy/blob/92ebe1e9a6aeb47a881a1226b08218175776f9ea/numpy/lib/histograms.py#L942-L1128) doesn't reuse code from `histogram()` either it looks like.

Reply:
Makes sense, thanks.